modules/i3s/src/lib/helpers/i3s-nodepages-tiles.ts

```diff
@@ -24,10 +34,11 @@ export default class I3SNodePagesTiles {
    * @param tileset - i3s tileset header ('layers/0')
    * @param options - i3s loader options
    */
-  constructor(tileset: I3sTilesetHeader, options: object) {
+  constructor(tileset: I3STilesetHeader, options: LoaderOptions) {
     this.tileset = {...tileset}; // spread the tileset to avoid circular reference
-    this.nodesPerPage = tileset.nodePages.nodesPerPage;
-    this.lodSelectionMetricType = tileset.nodePages.lodSelectionMetricType;
+    // @ts-ignore
+    this.nodesPerPage = tileset.nodePages?.nodesPerPage;
```

Comment:
I would not to ignore this error. Either define `nodesPerPage: 0` as meaning that nodePages are not supported, or throw an error, or...

There is a big value in the remaining code being able to rely on this field being a number. By typing it as a number, and then "secretly" setting it to "undefined", your typings could make more harm than good.

Reply:
Redone types here